src/couch/src/couch_bt_engine.erl

```diff
@@ -283,6 +340,31 @@ get_security(#st{header = Header} = St) ->
     end.
 
 
+get_props(#st{header = Header} = St) ->
+    case couch_bt_engine_header:get(Header, props_ptr) of
+        undefined ->
+            [];
+        Pointer ->
+            {ok, Props} = couch_file:pread_term(St#st.fd, Pointer),
+            Props
+    end.
+
+
+get_prop(St, Key) ->
+    Props = get_props(St),
+    case lists:keyfind(Key, 1, Props) of
+        false -> {error, no_value};
+        {Key, Value} -> {ok, Value}
+    end.
+
+
+get_prop(St, Key, DefaultValue) ->
+    case get_prop(St, Key) of
+        {error, no_value} -> DefaultValue;
+        Value -> Value
```

Comment:
There is a discrepancy in return value here. `get_prop/2` returns `{ok, Value}`. So we either change return value for `get_prop/2` or change `{error, no_value} -> {ok, DefaultValue}`.

Reply:
This was intentional but I see your point. get_prop/3 will always return a value whereas get_prop/2 might not, which is why the result shape is different. We could change get_prop/2 to return `Value` or `undefined` instead, which matches proplists:get_value behaviour.